src/seedu/addressbook/AddressBook.java

```diff
@@ -207,9 +207,20 @@
      */
 
     public static void main(String[] args) {
-        showWelcomeMessage();
-        processProgramArgs(args);
-        loadDataFromStorage();
+    	showToUser(DIVIDER, DIVIDER, VERSION, MESSAGE_WELCOME, DIVIDER);
+    	if (args.length >= 2) {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
+        }
+
+        if (args.length == 1) {
+            setupGivenFileForStorage(args[0]);
+        }
+
+        if(args.length == 0) {
+            setupDefaultFileForStorage();
+        }
+        initialiseAddressBookModel(loadPersonsFromFile(storageFilePath));
```

Comment:
Can make this code more worse if you try harder :) There are still method calls in between and use of constant variables.